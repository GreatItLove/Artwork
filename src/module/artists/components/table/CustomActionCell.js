import React, {Component} from 'react';
import FontIcon from 'material-ui/FontIcon';
import IconMenu from 'material-ui/IconMenu';
import MenuItem from 'material-ui/MenuItem';
import IconButton from 'material-ui/IconButton';
import PropTypes from 'prop-types';

import MoreVertIcon from 'material-ui/svg-icons/navigation/more-vert';
import {
  openUserDeleteDialog,
  openUserSuspendDialog,
  openUserEditDialog,
  openUserResetPasswordDialog,
} from '../../usersActions';
import RowDataHOC from '../../../common/HOCs/RowDataHOC';

const propTypes = {
  value: PropTypes.number.isRequired,
  appDispatch: PropTypes.func.isRequired,
  rowData: PropTypes.shape({
    active: PropTypes.bool.isRequired,
  }).isRequired,
};

/**
 * @class CustomActionCell
 * @extends React.Component
 * @description Render component
 */
class CustomActionCell extends Component {
  handleOpenDeleteDialog = () => {
    const {value} = this.props;
    /**
     * @see https://github.com/GriddleGriddle/Griddle/issues/647
     */
    this.props.appDispatch(openUserDeleteDialog(value));
  };

  handleOpenSuspendDialog = () => {
    const {value} = this.props;
    this.props.appDispatch(openUserSuspendDialog(value));
  };

  handleOpenEditUserDialog = () => {
    const {value} = this.props;
    this.props.appDispatch(openUserEditDialog(value));
  };

  handleOpenResetPasswordDialog = () => {
    const {value} = this.props;
    this.props.appDispatch(openUserResetPasswordDialog(value));
  };

  /**
   * @description render
   * @return {Object} JSX HTML Content
   */
  render() {
    const {rowData: {active}} = this.props;

    return (
      <div className="griddle-cell-action">
        <IconMenu
          iconButtonElement={
            <IconButton>
              <MoreVertIcon />
            </IconButton>
          }
          anchorOrigin={{horizontal: 'left', vertical: 'top'}}
          targetOrigin={{horizontal: 'left', vertical: 'top'}}>
          <MenuItem
            primaryText="Edit User"
            leftIcon={<FontIcon className="material-icons">create</FontIcon>}
            onClick={this.handleOpenEditUserDialog}
          />
          <MenuItem
            primaryText={active ? 'Suspend' : 'Restore'}
            leftIcon={
              <FontIcon className="material-icons">not_interested</FontIcon>
            }
            onClick={this.handleOpenSuspendDialog}
          />
          <MenuItem
            primaryText="Reset password"
            leftIcon={<FontIcon className="material-icons">lock</FontIcon>}
            onClick={this.handleOpenResetPasswordDialog}
          />
          <MenuItem
            primaryText="Delete"
            leftIcon={<FontIcon className="material-icons">delete</FontIcon>}
            onClick={this.handleOpenDeleteDialog}
          />
        </IconMenu>
      </div>
    );
  }
}

CustomActionCell.propTypes = propTypes;

export {CustomActionCell as CustomActionCellTest};

export default RowDataHOC(CustomActionCell);
