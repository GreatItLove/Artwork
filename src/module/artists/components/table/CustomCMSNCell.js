import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  value: PropTypes.string.isRequired,
};

/**
 * @function CustomActiveCell
 * @param {String} value
 * @returns {XML}
 */
function CustomCMSNCell({ value }) {
  const content = <span>{(value * 100).toFixed(2)} %</span>;
  return content;
}

CustomCMSNCell.propTypes = propTypes;

export default CustomCMSNCell;
