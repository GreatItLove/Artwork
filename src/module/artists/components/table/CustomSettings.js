import React from 'react';
import PropTypes from 'prop-types';
import RaisedButton from 'material-ui/RaisedButton';
import _map from 'lodash/map';

const propTypes = {
  appDispatch: PropTypes.func.isRequired,
  selectedCheckBox: PropTypes.array.isRequired,
  list: PropTypes.array.isRequired,
  fetching: PropTypes.bool.isRequired,
};

/**
 * @function CustomSettings
 * @param {object} selectedCheckBox checkedbox list
 * @returns {XML}
 */
class CustomSettings extends React.Component {
  changeColumnValue = settings => {
    const { appDispatch, bulkEditRequest, selectedCheckBox, id } = this.props;
    appDispatch(
      bulkEditRequest({
        [id]: selectedCheckBox.toString(),
        [settings.key]: settings.value,
      }),
    );
  };

  render() {
    const { t, settingsData } = this.props;
    const isDisabled =
      this.props.selectedCheckBox.length === 0 || this.props.fetching;
    const buttonStyle = isDisabled
      ? { borderRadius: '5px' }
      : { backgroundColor: '#E38925', borderRadius: '5px' };

    return this.props.list.length > 0 ? (
      <div className="filter-settings">
        <span style={{ color: 'lightgrey' }}>{t('bulkedits.comment')}</span>
        <div>
          {_map(settingsData, (settings, key) => (
            <RaisedButton
              key={key}
              label={t(`bulkedits.${settings.name}`)}
              primary={true}
              onClick={this.changeColumnValue.bind(this, settings)}
              buttonStyle={buttonStyle}
              labelStyle={{ textTransform: 'none' }}
              style={{ width: '140px', margin: '10px' }}
              disabled={isDisabled}
            />
          ))}
        </div>
      </div>
    ) : null;
  }
}

CustomSettings.propTypes = propTypes;

export default CustomSettings;
