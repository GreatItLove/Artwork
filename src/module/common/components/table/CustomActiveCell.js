import React from 'react';
import FontIcon from 'material-ui/FontIcon';
import PropTypes from 'prop-types';

const propTypes = {
  value: PropTypes.bool.isRequired
};

/**
 * @function CustomActiveCell
 * @param {String} value
 * @returns {XML}
 */
function CustomActiveCell({ value }) {
  const content = value ? (
    <FontIcon className="material-icons">done</FontIcon>
  ) : (
    <FontIcon className="material-icons">clear</FontIcon>
  );
  return content;
}

CustomActiveCell.propTypes = propTypes;

export default CustomActiveCell;
