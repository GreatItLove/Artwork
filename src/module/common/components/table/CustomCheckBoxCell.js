import React from 'react';
import PropTypes from 'prop-types';
import CheckBox from '../form/CheckBox';

const propTypes = {
  rowData: PropTypes.object.isRequired,
  appDispatch: PropTypes.func.isRequired,
  checkAction: PropTypes.func.isRequired,
  unCheckAction: PropTypes.func.isRequired,
  id: PropTypes.string.isRequired,
};

class CustomCheckBoxCell extends React.Component {
  onChange = e => {
    const { rowData, appDispatch, checkAction, unCheckAction, id } = this.props;
    if (rowData.checked) appDispatch(unCheckAction(rowData[id]));
    else appDispatch(checkAction(rowData[id]));
  };

  render() {
    const { rowData } = this.props;
    return (
      <div style={{ width: "20px" }}>
        <CheckBox iconStyle={rowData.checked ? { fill: '#E38925' } : null} input={{ value: rowData.checked, onChange: this.onChange }} label="" />
      </div>
    )
  }
}

CustomCheckBoxCell.propTypes = propTypes;

export default CustomCheckBoxCell;
