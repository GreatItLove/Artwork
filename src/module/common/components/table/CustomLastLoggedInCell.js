import React from 'react';
import PropTypes from 'prop-types';

import convertTimestamp from '../../../../utility/convertTimestamp';

const propTypes = {
  value: PropTypes.string.isRequired
};

/**
 * @function CustomLastLoggedInCell
 * @param {Number} value
 * @returns {XML}
 */
function CustomLastLoggedInCell({ value }) {
  return <span>{value && convertTimestamp(value)}</span>;
}

CustomLastLoggedInCell.propTypes = propTypes;
export default CustomLastLoggedInCell;
