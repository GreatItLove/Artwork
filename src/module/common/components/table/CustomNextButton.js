import React from 'react';
import PropTypes from 'prop-types';
import { RaisedButton } from 'material-ui';
import HardwareKeyboardArrowRight from 'material-ui/svg-icons/hardware/keyboard-arrow-right';

const propTypes = {
  className: PropTypes.string,
  style: PropTypes.object,
  hasNext: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired,
};

/**
 * @function CustomNextButton
 * @param {String} className
 * @param {Object} style
 * @param {Boolean} hasNext
 * @param {Function} onClick
 * @returns {XML}
 */
function CustomNextButton({ className, style, hasNext, onClick }) {
  const styles = {
    ...style,
    minWidth: '36px',
  }
  return (
    <RaisedButton
      className={className}
      disabled={!hasNext}
      onClick={onClick}
      backgroundColor="#f2f2f2"
      style={styles}
      buttonStyle={{ cursor: !hasNext ? 'not-allowed' : 'pointer', width: '36px' }}
      icon={
        <HardwareKeyboardArrowRight
          color="#e18927"
          style={{ height: 40, width: 40 }}
        />}
    />
  );
}

CustomNextButton.propTypes = propTypes;

export default CustomNextButton;
