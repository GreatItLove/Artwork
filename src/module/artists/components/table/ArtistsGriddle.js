import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import Griddle, { RowDefinition, ColumnDefinition } from 'griddle-react';
import PropTypes from 'prop-types';
import _map from 'lodash/map';
import { RefreshIndicator } from 'material-ui';

import CustomSettings from './CustomSettings';
import CustomCheckBoxCell from '../../../common/components/table/CustomCheckBoxCell';
import CustomActiveCell from '../../../common/components/table/CustomActiveCell';
import CustomNextButton from '../../../common/components/table/CustomNextButton';
import CustomPreviousButton from '../../../common/components/table/CustomPreviousButton';
import CustomShowPerPage from '../../../common/components/table/CustomShowPerPage';
import CustomFilter from '../../../common/components/filter/CustomFilter';
import RowData from '../../../common/HOCs/RowDataHOC';

import CustomCMSNCell from './CustomCMSNCell';
import AddArtistDialog from '../../containers/AddArtistDialog';
import DownloadArtistDialog from '../../containers/DownloadArtistDialog';
import { ARTISTS_EDIT_PATH } from '../../../../config/routesPatch';
import {
  fetchArtistsRequest,
  updateSearchFilter,
  switchArtistFilter,
  checkArtists,
  uncheckArtists,
  bulkArtistEditRequest,
} from '../../artistsActions';

const propTypes = {
  list: PropTypes.array.isRequired,
  appDispatch: PropTypes.func.isRequired
};

const tableStyleConfig = {
  classNames: {
    Table: 'table table-hover',
  }
};

function FullNameActiveCell({ value, griddleKey, rowData }) {
  return (
    <div>
      <NavLink to=/*{`${ARTISTS_EDIT_PATH}/${rowData.artistsId}`}*/{ARTISTS_EDIT_PATH + "/" + rowData.artistsId}> {rowData.firstName} {rowData.lastName} </NavLink>
    </div>
  );
}

const NewLayout = ({ Table, Pagination, Filter, SettingsWrapper }) => (
  <div>
    <Filter />
    <Table />
    <SettingsWrapper />
    <Pagination />
  </div>
);

/*
 * @class UsersGriddle
 * @extends React.Component
 * @description Render component
 */
class ArtistsGriddle extends Component {
  /**
   * @description render
   * @return {Object} JSX HTML Content
   */

  constructor(props) {
    super(props);
    this._onSort = this._onSort.bind(this);
    this._onFilter = this._onFilter.bind(this);
    this._onNext = this._onNext.bind(this);
    this._onPrevious = this._onPrevious.bind(this);
    this._onGetPage = this._onGetPage.bind(this);

    const { appDispatch, list } = this.props;
    if (list.length === 0)
      appDispatch(fetchArtistsRequest({ limit: 25, offset: 0 }));

  }

  _onFilter(filterText) {
    const { appDispatch, pagination } = this.props;
    appDispatch(updateSearchFilter(filterText));
    appDispatch(fetchArtistsRequest({ limit: pagination.recordsPerPage, ...filterText }));
  }
  _onSort(sortProperties) {
  }

  _onNext() {
    const { appDispatch, pagination } = this.props;
    appDispatch(fetchArtistsRequest({ limit: pagination.recordsPerPage, offset: pagination.currentPage * pagination.recordsPerPage, ...this.props.searchFilter }));
  }

  _onPrevious() {
    const { appDispatch, pagination } = this.props;
    appDispatch(fetchArtistsRequest({ limit: pagination.recordsPerPage, offset: (pagination.currentPage - 2) * pagination.recordsPerPage, ...this.props.searchFilter }));
  }

  _onGetPage(pageNumber) {
    const { appDispatch, pagination } = this.props;
    appDispatch(fetchArtistsRequest({ limit: pagination.recordsPerPage, offset: (pageNumber - 1) * pagination.recordsPerPage, ...this.props.searchFilter }));
  }

  render() {
    const { t, list, appDispatch, pagination, selectedCheckBox, genres } = this.props;
    let self = this;
    for (var i = 0; i < list.length; i++) {
      if (selectedCheckBox.indexOf(list[i].artistsId) >= 0) {
        list[i].checked = true;
      } else list[i].checked = false;
    }
    let fieldValue = [];
    _map(genres, data => {
      fieldValue.push({ name: data, value: data });
    });
    let filterData = {
      name: 'Artists',
      radio: [
        {
          name: 'active',
          fields: [{ name: 'Yes', value: true }, { name: 'No', value: false }],
        },
        {
          name: 'featured',
          fields: [{ name: 'Yes', value: true }, { name: 'No', value: false }],
        },
        {
          name: 'artworkCount',
          fields: [{ name: 'Yes', value: 1 }, { name: 'No', value: 0 }],
        },
        {
          name: 'genre',
          fields: fieldValue,
        },
      ],
    };
    const settingsData = [
      { name: 'active', key: 'active', value: true },
      { name: 'inactive', key: 'active', value: false },
      { name: 'featured', key: 'featured', value: true },
      { name: 'unfeatured', key: 'featured', value: false },
    ];
    const extraData = {
      appDispatch: appDispatch,
      checkAction: checkArtists,
      unCheckAction: uncheckArtists,
      id: 'artistsId',
    };
    return (
      <div>
        <Griddle
          data={list}
          styleConfig={tableStyleConfig}
          pageProperties={{
            currentPage: pagination.currentPage,
            pageSize: pagination.recordsPerPage,
            recordCount: pagination.totalRecordCount,
          }}
          events={{
            onFilter: this._onFilter,
            onSort: this._onSort,
            onNext: this._onNext,
            onPrevious: this._onPrevious,
            onGetPage: this._onGetPage,
          }}
          components={{
            SettingsWrapper: () => {
              return (
                <CustomSettings
                  t={t}
                  id="artistsIds"
                  appDispatch={self.props.appDispatch}
                  selectedCheckBox={self.props.selectedCheckBox}
                  list={self.props.list}
                  settingsData={settingsData}
                  fetching={self.props.fetching}
                  bulkEditRequest={bulkArtistEditRequest}
                />
              )
            },
            Filter: filter => {
              return (
                <CustomFilter
                  t={t}
                  showingFilter={self.props.showingFilter}
                  searchFilter={self.props.searchFilter}
                  recordCount={self.props.pagination.totalRecordCount}
                  filterData={filterData}
                  switchFilter={switchArtistFilter}
                  setFilter={filter.setFilter}
                  className={filter.className}
                  appDispatch={appDispatch}
                  style={filter.style}
                />
              )
            },
            PageDropdown: () => null,
            NextButton: CustomNextButton,
            PreviousButton: CustomPreviousButton,
            Layout: NewLayout,
            Table: ({ className, style, visibleRows, TableHeading, TableBody, NoResults }) => {
              return (self.props.fetching ?
                <div className={className} style={{ position: 'absolute', bottom: '120px', left: '48%', ...style }}>
                  <RefreshIndicator left={0} top={0} status="loading" />
                </div>
                :
                visibleRows ?
                  <table className={className} style={style}>
                    <TableHeading />
                    <TableBody />
                  </table>
                  :
                  <NoResults />

              )
            }
          }}>
          <RowDefinition>
            <ColumnDefinition id="check" title=" "
              customComponent={RowData(CustomCheckBoxCell)}
              extraData={extraData}
            />
            <ColumnDefinition id="firstName" title={t("columns.name")} customComponent={RowData(FullNameActiveCell)} />
            <ColumnDefinition id="artworkCount" title={t("columns.art")} />
            <ColumnDefinition id="icon" title={t("columns.icon")} />
            <ColumnDefinition id="genre" title={t("columns.genre")} />
            <ColumnDefinition id="commission" title={t("columns.cmsn")} customComponent={CustomCMSNCell} />
            <ColumnDefinition id="taxNumber" title={t("columns.taxNumber")} />
            <ColumnDefinition id="featured" title={t("columns.featured")} customComponent={CustomActiveCell} />
            <ColumnDefinition id="active" title={t("columns.active")} customComponent={CustomActiveCell} />
          </RowDefinition>
        </Griddle>
        <CustomShowPerPage fetchRequest={fetchArtistsRequest} {...this.props} />
        <AddArtistDialog />
        <DownloadArtistDialog />
      </div>
    );
  }
}

ArtistsGriddle.propTypes = propTypes;

export default ArtistsGriddle;
