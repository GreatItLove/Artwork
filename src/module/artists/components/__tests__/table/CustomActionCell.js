import React from 'react';
import {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';

import {CustomActionCellTest as CustomActionCell} from '../../table/CustomActionCell';
describe('Custom Action Cell', () => {
  it('Snapshot of CustomActionCell', () => {
    const props = {
      value: 156654,
      appDispatch: jest.fn(),
      rowData: {
        active: true,
      },
    };
    const wrapper = shallow(<CustomActionCell {...props} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
