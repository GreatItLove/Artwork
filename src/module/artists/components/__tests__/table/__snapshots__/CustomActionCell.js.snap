// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Action Cell Snapshot of CustomActionCell 1`] = `
<div
  className="griddle-cell-action"
>
  <IconMenu
    anchorOrigin={
      Object {
        "horizontal": "left",
        "vertical": "top",
      }
    }
    animated={true}
    iconButtonElement={
      <IconButton
        disableTouchRipple={false}
        disabled={false}
        iconStyle={Object {}}
        tooltipPosition="bottom-center"
        touch={false}
      >
        <NavigationMoreVert />
      </IconButton>
    }
    multiple={false}
    onClick={[Function]}
    onItemTouchTap={[Function]}
    onKeyboardFocus={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onRequestChange={[Function]}
    open={null}
    targetOrigin={
      Object {
        "horizontal": "left",
        "vertical": "top",
      }
    }
    touchTapCloseDelay={200}
    useLayerForClickAway={false}
  >
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      leftIcon={
        <FontIcon
          className="material-icons"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          create
        </FontIcon>
      }
      onClick={[Function]}
      primaryText="Edit User"
      targetOrigin={
        Object {
          "horizontal": "left",
          "vertical": "top",
        }
      }
    />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      leftIcon={
        <FontIcon
          className="material-icons"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          not_interested
        </FontIcon>
      }
      onClick={[Function]}
      primaryText="Suspend"
      targetOrigin={
        Object {
          "horizontal": "left",
          "vertical": "top",
        }
      }
    />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      leftIcon={
        <FontIcon
          className="material-icons"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          lock
        </FontIcon>
      }
      onClick={[Function]}
      primaryText="Reset password"
      targetOrigin={
        Object {
          "horizontal": "left",
          "vertical": "top",
        }
      }
    />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      leftIcon={
        <FontIcon
          className="material-icons"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          delete
        </FontIcon>
      }
      onClick={[Function]}
      primaryText="Delete"
      targetOrigin={
        Object {
          "horizontal": "left",
          "vertical": "top",
        }
      }
    />
  </IconMenu>
</div>
`;
